### DISCUSSION TITLE
combining filament form fields with existing forms

### DISCUSSION THREAD

**Discussion by mokhosh** [filamentphp/filament#1378] [Help] [answered]
I have a fully functioning web application, and the only thing I want to change is make my rich text editors able to accept image uploads, and one of the cleanest implementations i found was filament.
but the problem is when i add the interface and trait, my validation rules are ignored and previous form elements stop working.
is there an easy way to combine filament form elements with existing livewire forms?

**Comment by danharrin** [comment, marked as answer]
Hey, the `InteractsWithForms` trait overrides `rules()`, so you'd need to import the trait method and merge it with your custom rules. But I would always recommend replacing your whole form with Filament is you want a cleaner solution ;)

https://andy-carter.com/blog/overriding-extending-a-php-trait-method
